widgets/src/layout/expanded.rs

```diff
@@ -4,15 +4,51 @@ use ribir_core::prelude::*;
 /// available space. If multiple children are expanded, the available space is
 /// divided among them according to the flex factor.
 #[derive(Clone, PartialEq)]
-// `Expand` should not support `FatObj`, as this may cause the `Expanded` to be invisible to its
-// parent. `@Expanded { margin: EdgeInsets::all(10.) }` actually expands as `@Margin { @Expanded {
-// .. } }`.
-#[simple_declare]
+// `Expand` should not support `FatObj`, as this may cause the `Expanded` to be
+// invisible to its parent. `@Expanded { margin: EdgeInsets::all(10.) }`
+// actually expands as `@Margin { @Expanded { .. } }`.
 pub struct Expanded {
-  #[declare(default = 1.)]
   pub flex: f32,
 }
 
+impl Default for Expanded {
+  fn default() -> Self { Self { flex: 1. } }
+}
+
+#[derive(Default)]
+pub struct FlexDeclarer {
```

Comment:
Prefer using `ExpandedDeclarer` as `FlexDeclarer` could be ambiguous with the declarer of `Flex`.